File: 1.two-sum.php
/*
 * @lc app=leetcode id=1 lang=php
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        asort($nums);
        $searchArray = $nums;
        for ($i = 0; $i < count($nums); $i++) {
            $complement = $target - $searchArray[$i];
            unset($searchArray[$i]);
            if (in_array($complement, $searchArray)) {
                $key = array_keys($searchArray, $complement);
                return [$i, $key[0]];
            }
        }
    }

}
// @lc code=end

File: 1.two-sum.rb
#
# @lc app=leetcode id=1 lang=ruby
#
# [1] Two Sum
#

# @lc code=start
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    flip = nums.map.with_index.to_h
    p flip
    nums.each_with_index{|value,index|
        diff = target - value
        if flip[diff] && flip[diff] != index
            return [index, flip[diff]];
        end
    }
end

# @lc code=end

